conftest.py

```diff
@@ -11,20 +11,72 @@
 ###############################################################################
 
 import sys
+import os
+import logging
+
+import pytest
+from typing import Mapping
+from hashlib import md5
 
 from spinalcordtoolbox.utils import sct_dir_local_path, sct_test_path
 sys.path.append(sct_dir_local_path('scripts'))
 
-import pytest
 import sct_download_data as downloader
 
 
-@pytest.fixture(scope='session', autouse=True)
-def download_data(request):
-    # This is a hack because the capsys fixture can't be used with
-    # session scope at the moment.
-    # https://github.com/pytest-dev/pytest/issues/2704#issuecomment-603387680
-    capmanager = request.config.pluginmanager.getplugin("capturemanager")
-    with capmanager.global_and_fixture_disabled():
-        print('\nDownloading sct testing data.')
-        downloader.main(['-d', 'sct_testing_data', '-o', sct_test_path()])
+logger = logging.getLogger(__name__)
+
+
+def pytest_sessionstart():
+    """ Download sct_testing_data prior to test collection. """
+    logger.info("Downloading sct test data")
+    downloader.main(['-d', 'sct_testing_data', '-o', sct_test_path()])
+
+
+@pytest.fixture(scope="session", autouse=True)
+def test_data_integrity(request):
+    files_checksums = dict()
+    for root, _, files in os.walk(sct_test_path()):
+        for f in files:
+            fname = os.path.join(root, f)
+            chksum = checksum(fname)
+            files_checksums[fname] = chksum
+
+    request.addfinalizer(lambda: check_testing_data_integrity(files_checksums))
+
+
+def checksum(fname: os.PathLike) -> str:
+    with open(fname, 'rb') as f:
+        data = f.read()
+    return md5(data).hexdigest()
```

Comment:
Just out of curiosity, for my own learning, why `md5.hexdigest`? (as opposed to other options provided by `hashlib`)

Reply:
No particular reason. `md5` is a fast (128bits) and popular hashing function. It shouldn't be used for security applications anymore, but for what we're doing here it's ok. Honestly I didn't really evaluate other options.

Reply:
That makes sense to me! For something like this, I agree, "perfect is the enemy of the good", etc. :slightly_smiling_face: